Update Styling the content article (#39061)
mdn/content: files/en-us/learn_web_development/getting_started/your_first_website/styling_the_content/index.md

{{PreviousMenuNext("Learn_web_development/Getting_started/Your_first_website/Creating_the_content", "Learn_web_development/Getting_started/Your_first_website/Adding_interactivity", "Learn_web_development/Getting_started/Your_first_website")}}

> CSS (Cascading Style Sheets) is the code that styles web content. This article walks through a basic understanding of CSS and its functionality and shows you how to improve the look and feel of the content structure you created in the previous article.

Author: updated!

## What is CSS?

> Like HTML, CSS is not a programming language. It's not a markup language either. **CSS is a style sheet language.** CSS is used to style HTML elements; you select which elements you want to style, and set values for their style properties that define how they will look.

Author: Nice; updated.

> Let's revisit the HTML example we saw at the start of [Creating the content](/en-US/docs/Learn_web_development/Getting_started/Your_first_website/Creating_the_content):

Author: updated, with a small tweak:

> Let's revisit the basic HTML example from the article [Creating the content](/en-US/docs/Learn_web_development/Getting_started/Your_first_website/Creating_the_content):

> Let's revisit the HTML example we saw at the start of [Creating the content](/en-US/docs/Learn_web_development/Getting_started/Your_first_website/Creating_the_content):

Reviewer: okay, sure

```html live-sample___basic-html live-sample___basic-css
<p>Instructions for life:</p>

<ul>
  <li>Eat</li>
  <li>Sleep</li>
  <li>Repeat</li>
</ul>
```

This renders as follows on its own:

{{EmbedLiveSample("basic-html", "100%", "140px")}}

If we add some CSS into the mix, we can change how the HTML looks. The following snippet selects the {{htmlelement("p")}} element and gives it a different [font](/en-US/docs/Web/CSS/font-family) and a red text {{cssxref("color")}}. It then selects all of the {{htmlelement("li")}} elements and gives each one a yellow {{cssxref("background-color")}}, a one-pixel solid black {{cssxref("border")}}, and a 5-pixel [bottom margin](/en-US/docs/Web/CSS/margin-bottom):

```css live-sample___basic-css
p {
  font-family: sans-serif;
  color: red;
}

li {

>   background-color: yellow;

Author: Hrm, you don't like my loud demos? ;-)

No matter. I have changed the color to `greenyellow`, which is a bit more subtle, and updated the description to suit. I wanted to keep the color value as an obvious named color rather than a weird value or a hex value, as that's one less thing to worry about explaining at this stage.

>   background-color: yellow;

Reviewer: Yeah, that is a valid point to use a color with an uncomplicated name.

`greenyellow` is slightly better but still loud for me ;) But it is accessible, so let's go with it

  border: 1px solid black;
  margin-bottom: 5px;
}
```

With the CSS applied to the HTML, the demo now renders like this:

{{EmbedLiveSample("basic-css", "100%", "160px")}}

> 

Author: Yeah, I like it — added.

> CSS has many other functions, from specifying background images and gradients and precisely controlling typography and scrolling behavior, to implementing animations and entire web page layouts!

Author: updated

## Applying CSS to your HTML

When using CSS, the first thing to get right is making sure that your CSS is successfully applied to your HTML. In this section we will add a CSS **stylesheet** to your `first-website`, and apply it to your HTML.

1. Inside your `first-website` folder, create another new folder called `styles`.
2. Using a text editor, paste the following CSS into a new file, which will give your `<p>` elements a red text color. It is useful to insert something like this into your stylesheet to begin with, to test whether it is being applied to your HTML correctly.

   ```css
   p {
     color: red;
   }
   ```

> 3. Save the file inside your `styles` folder with a filename of `style.css`.

Author: updated

4. Open your `index.html` file. Paste the following line inside the HTML head (between the {{HTMLElement("head")}} and `</head>` tags):

   ```html
   <link href="styles/style.css" rel="stylesheet" />
   ```

5. Save `index.html` and load it in your browser. You should see something like this:

![A Mozilla logo and some paragraphs. The paragraph text has been styled red by our css.](website-screenshot-styled.png)

If your paragraph text is red, congratulations! Your CSS is working. If not, go through the above steps and check carefully that you have followed each one correctly.

> ## A brief aside on CSS syntax

Reviewer: ```suggestion
## CSS syntax basics
```
It looks like this can be moved to before the "Applying CSS to your HTML" section.

Author: I agree with the title change, but I don't want to move it before the previous section because it refers to the code from it.

> In the above CSS, the `p` is called a **selector** — it selects which element(s) to style, in this case all the paragraphs in the HTML it is applied to. The line inside the curly braces (`{ }`) is called a **declaration**, which sets a new value for a specific **property** (in this case, `color`, which represents the text color of the paragraphs); the **property value** set is `red`.

Reviewer: slight tweaks for consistency
```suggestion
In the previous CSS example, `p` is called a **selector** — it selects the element(s) to style; in this case, all the paragraphs in the HTML. The line inside the curly braces (`{ }`) is called a **declaration** – it sets a value for a specific property. In this case, the **property** is `color`, which controls the text color of the paragraphs, and the **property value** set is `red`.
```

Reviewer: Yeah, your tweak makes it better

}
```

> Within a ruleset, you must use a semicolon (`;`) to separate each declaration from the next one. Within each declaration, you must use a colon (`:`) to separate the property from its value.

Reviewer: ```suggestion
Within a ruleset, you must use a semicolon (`;`) to separate one declaration from the next. Within each declaration, you must use a colon (`:`) to separate the property and its value.
```

Author: updated

> You can also include multiple selectors on one rule, separated by commas, to select multiple elements. For example:

Reviewer: ```suggestion
You can also include multiple selectors in one rule, separated by commas, to select multiple elements. For example:
```

Author: Good catch; fixed.

```css
p,
.my-class,
#my-id {
  color: red;
}
```

> In the above CSS rule we have included an **element** (or **type**) selector, which selects a specific HTML element, but we have also included two other selector types, which aren't relevant to the rest of this tutorial. If you are curious as to what they do, check out our [Basic selectors](/en-US/docs/Learn_web_development/Core/Styling_basics/Basic_selectors) guide to find out.

Reviewer: ```suggestion
In this CSS rule, we've included an **element** (or **type**) selector, which selects a specific HTML element. (We've also included two other selector types, which aren't relevant to the rest of this tutorial. If you're curious about what they do, check out our [Basic selectors](/en-US/docs/Learn_web_development/Core/Styling_basics/Basic_selectors) guide.)
```

Author: I've updated it, but I've also gotten rid of the parens. I didn't think they were needed.

> [!NOTE]
> Scrimba's [Write your first lines of CSS!](https://v2.scrimba.com/the-frontend-developer-career-path-c0j/~015?via=mdn) <sup>[_MDN learning partner_](/en-US/docs/MDN/Writing_guidelines/Learning_content#partner_links_and_embeds)</sup> provides a useful interactive introduction to CSS syntax.

> ## Fonts and text

Author: I've updated it to the more general "Improving the text", as this isn't inaccurate. WDYT?

> ## Fonts and text

Reviewer: wfm, thanks!

> Let's return to our example and use CSS to improve the appearance of the text — we'll set a new font for the page and change some text settings on the different elements.

Reviewer: ```suggestion
Let's return to our example and use CSS to improve the appearance of the text. We'll set a new font for the page and change some text settings for different elements.
```

Author: updated

1. First, find the [output from Google Fonts](/en-US/docs/Learn_web_development/Getting_started/Your_first_website/What_will_your_website_look_like#choosing_a_font) that you previously saved. If you've not already chosen a font, follow the link and do it now.
2. Add the {{htmlelement("link")}} elements inside your `index.html`'s {{HTMLElement("head")}}, just before the closing `</head>` tag. They should look something like this:

   ```html
   <link rel="preconnect" href="https://fonts.googleapis.com" />
   <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
   <link
     href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100..900;1,100..900&display=swap"
     rel="stylesheet" />
   ```

>    This code links your page to a style sheet on the Google Font service that loads your chosen font.

Reviewer: ```suggestion
   This code links your page to a stylesheet hosted by the Google Fonts service, which loads your chosen font.
```

Author: updated

3. Next, go to your `style.css` file and delete the existing rule. We no longer want our paragraphs to be red.
4. Add the following lines to `style.css`:

   ```css
   html {

>      /* px means "pixels": the base font size is now 10 pixels high */

Reviewer: ```suggestion
     /* px means "pixels". The base font size is now 10 pixels high */
```

     font-size: 10px;

>      /* This should be replaced with the font-family output you got from Google Fonts */

Reviewer: ```suggestion
     /* Replace PLACEHOLDER with the font-family output you got from Google Fonts */
```

Reviewer: Yeah, this is better

     font-family: PLACEHOLDER;
   }
   ```

   > [!NOTE]
   > Anything in CSS between `/*` and `*/` is a **CSS comment**, which is ignored by the browser. CSS comments are a way for you to include helpful notes about your code or logic, without affecting how your webpage is rendered.

5. Replace the `font-family` placeholder line with the `font-family` line in your Google Fonts code, for example:

   ```css
   font-family: "Roboto", sans-serif;
   ```

   The `font-family` property sets the font(s) you want to apply to your HTML. This rule defines a global base font and font size for the whole page. All elements inside the {{HTMLElement("html")}} element will inherit the same `font-size` and `font-family`.

> 6. Now let's set some font and text styles on our [`<h1>`](/en-US/docs/Web/HTML/Element/Heading_Elements), {{htmlelement("li")}}, and {{htmlelement("p")}} elements. Here we set new {{cssxref("font-size")}} values for each element. We also center the heading using {{cssxref("text-align")}} and increase the {{cssxref("line-height")}} and {{cssxref("letter-spacing")}} of the paragraphs and list items to make the body content more readable.

Reviewer: ```suggestion
6. Now let's set some font and text styles on our [`<h1>`](/en-US/docs/Web/HTML/Element/Heading_Elements), {{htmlelement("li")}}, and {{htmlelement("p")}} elements. We'll set new {{cssxref("font-size")}} values for each element. We'll also center the heading using {{cssxref("text-align")}} and increase the {{cssxref("line-height")}} and {{cssxref("letter-spacing")}} of the paragraphs and list items to make the body content more readable.
```

Author: updated

   }
   ```

7. Save your code and load your HTML in a browser (refresh it if you've got it open from before). Your work-in-progress should look similar to this:

   ![A Mozilla logo and some paragraphs. A sans-serif font has been set, the font sizes, line height and letter spacing are adjusted, and the main page heading has been centered](website-screenshot-font-small.png)

   > [!NOTE]
   > Try adjusting the `px` values until you end up with font sizes that you like for your heading and body text.

> ## CSS: all about boxes

Reviewer: or this?
```suggestion
## Understanding boxes in CSS
```

Author: I want to keep this title ;-)

Reviewer: We could keep it as is. I know that the suggested "understanding" one is quite boring and doesn't add the same effect as "all about boxes".

I am mostly thinking about how it fits into the flow of the page TOC - it shouldn't feel out of place or break the flow (just looking at the TOC). 
While we can go with what we have now, here's a few more options, closer to the original, you can possibly consider:
- "CSS is all about boxes"
- "Everything in CSS is a box"
- "Thinking in terms of boxes"
- "How CSS is all about boxes"
I guess I am essentially proposing avoiding the colon construction

Author: I'm happy to avoid the colon. I've updated it to "CSS is all about boxes"

Reviewer: Cool, looks good. Thanks

> Something you'll notice about CSS as you use it more is that a lot of it is about boxes. Most HTML elements on your page can be thought of as boxes sitting on top of (or alongside) other boxes, which you will then set values on for sizing, colors, positioning information, etc. This is referred to as **the box model**.

Reviewer: ```suggestion
Something you'll notice about CSS as you use it more is that a lot of it is about boxes. Most HTML elements on a page can be thought of as boxes that sit on top of (or alongside) other boxes. You can set values on these boxes for size, color, and positioning. This is referred to as **the box model**.
```
link to https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_box_model/Introduction_to_the_CSS_box_model?

Reviewer: Wondering what you think about the other suggested edits, especially splitting the sentence at "which"

Author: oops, I missed your suggestion on this one! I've used it as it was good.

Reviewer: thanks!

![Three boxes sat inside one another. From outside to in they are labelled margin, border and padding](box-model.png)

> Each box taking up space on your page has properties like:

Reviewer: ```suggestion
Each box that takes up space on your page has properties like:
```

Author: updated

> - {{cssxref("padding")}}: The space around the content. In the example below, it is the space around the paragraph text.
> - {{cssxref("border")}}: The solid line that is just outside the padding.
> - {{cssxref("margin")}}, the space around the outside of the border.

Reviewer: ```suggestion
- {{cssxref("padding")}}: The space around the content. In the previous example, it is the space around the paragraph text.
- {{cssxref("border")}}: The solid line just outside the padding.
- {{cssxref("margin")}}: The space outside the border.
```

Author: updated

> In this section we also use the following properties, some of which you've met before:

Reviewer: ```suggestion
In this section, we also use the following properties, some of which you've seen before:
```

- {{cssxref("width")}}: The width of an element.
- {{cssxref("background-color")}}: The color behind an element's content and padding.
- {{cssxref("color")}}: The color of an element's content (usually text).

> - {{cssxref("text-shadow")}}: Sets a drop shadow on the text inside an element.
> - {{cssxref("display")}}: Sets the display mode of an element (which basically refers to how it appears or is laid out inside the webpage).

Reviewer: ```suggestion
- {{cssxref("text-shadow")}}: A drop shadow on the text inside an element.
- {{cssxref("display")}}: The display mode of an element (which basically refers to how it appears or is laid out on the web page).
```

Author: Updated

> In each of the below sections:

Reviewer: ```suggestion
In each of the sections that follow:
```

Author: updated

1. Add the provided CSS code to the bottom of your `style.css` file.
2. Save the file and refresh your browser to see how the CSS has affected the HTML rendering.
3. Read the provided explanation to help you understand how the CSS works.

> 4. If you are feeling brave, experiment with changing the property values to further customize your page.

Reviewer: can be made to sound more encouraging :)
```suggestion
4. If you are feeling adventurous, experiment with changing the property values to further customize your page.
```

> ### Changing the page color

Author: I've tried it in my next commit; there are quite a lot of H2s in the TOC now, but I think it looks Ok. What do you think?

> ### Changing the page color

Reviewer: I like it :). Thanks

The above code sets new values for several properties of the {{htmlelement("body")}} element. Let's go through these line-by-line:

- `width: 600px;`: This forces the body to always be 600 pixels wide.
- `margin: 0 auto;`: When you set two values on a property like `margin` or `padding`, the first value affects the element's top _and_ bottom side (setting it to `0` in this case); the second value affects the left _and_ right side. `auto` is a special value that divides the available horizontal space evenly between left and right.

> - `background-color: #FF9500;`: This sets the element's background color. Our project uses a reddish orange for the body background color, as opposed to dark blue for the {{htmlelement("html")}} element.

Reviewer: ```suggestion
- `background-color: #FF9500;`: This sets the element's background color. Our project uses a reddish orange for the body background color, instead of the dark blue used for the {{htmlelement("html")}} element.
```

Reviewer: This update makes more sense. Thanks!

> - `padding: 0 20px 20px 20px;`: This sets four values for padding. The goal is to put some space around the content. In this example, there is no padding on the top of the body, and 20 pixels on the right, bottom and left. The values set top, right, bottom, and left padding, in that order.

Reviewer: ```suggestion
- `padding: 0 20px 20px 20px;`: This sets four values for padding. The goal is to put some space around the content. In this example, there is no padding on the top of the body, and 20 pixels on the right, bottom, and left. The values set top, right, bottom, and left padding, in that order.
```

Author: updated

> - `border: 5px solid black;`: This sets values for the width, style and color of the border. In this case, it's a five-pixel–wide, solid black border, on all sides of the body.

Reviewer: ```suggestion
- `border: 5px solid black;`: This sets values for the width, style, and color of the border. In this case, it's a 5-pixel-wide solid black border around all sides of the body.
```

Author: updated

}
```

Next, we center the image to make it look better. We can use the same `margin: 0 auto` trick as we did for the body, but there are differences that require an additional setting to make the CSS work.

> The {{htmlelement("body")}} element is a **block** element, meaning it takes up space on the page and can accept margin, padding, etc. {{htmlelement("img")}} (image) elements, on the other hand, are **inline** elements: by default, they don't accept margin values like block elements do. For the auto-margin trick to work on this image, we must give it block-level behavior using `display: block;`.

Reviewer: ```suggestion
The {{htmlelement("body")}} element is a **block** element, meaning it takes up space on the page and can accept margin, padding, and other box properties. {{htmlelement("img")}} (image) elements, on the other hand, are **inline** elements: by default, they don't accept margin values in the same way block elements do. For the auto-margin trick to work on this image, we must give it block-level behavior by using `display: block;`.
```

> Finally, we set the {{cssxref("max-width")}} property to `100%` to make sure that, if the image is larger than the `width` set on the body (600 pixels), it will be constained to `600px` and no wider.

Author: updated

> [!NOTE]
> Don't be too concerned if you don't completely understand `display: block;` and the differences between a block element and an inline element, or `max-width: 100%;`. They will make more sense as you continue your study of CSS.

## Conclusion

If you followed all the instructions in this article, you should have a page that looks similar to this one:

![A Mozilla logo, centered, and a header and paragraphs. It now looks nicely styled, with a blue background for the whole page and orange background for the centered main content strip.](website-screenshot-final.png)

You can [view our version here](https://mdn.github.io/beginner-html-site-styled/). If you get stuck, you can always compare your work with our [finished example code on GitHub](https://github.com/mdn/beginner-html-site-styled/blob/gh-pages/styles/style.css).

> Here, we have just scratched the surface of CSS. You'll learn a lot more in our [CSS styling basics](/en-US/docs/Learn_web_development/Core/Styling_basics) Core module later on in the course.

Reviewer: reverting to the original "in this article" :)
```suggestion
In this article, we've only scratched the surface of CSS. You'll learn a lot more in our [CSS styling basics](/en-US/docs/Learn_web_development/Core/Styling_basics) Core module later in the course.
```

Author: updated
